machine_learning_hep/validation_candidates.py

```diff
@@ -0,0 +1,56 @@
+#############################################################################
+##  © Copyright CERN 2018. All rights not expressly granted are reserved.  ##
+##                 Author: Gian.Michele.Innocenti@cern.ch                  ##
+## This program is free software: you can redistribute it and/or modify it ##
+##  under the terms of the GNU General Public License as published by the  ##
+## Free Software Foundation, either version 3 of the License, or (at your  ##
+## option) any later version. This program is distributed in the hope that ##
+##  it will be useful, but WITHOUT ANY WARRANTY; without even the implied  ##
+##     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    ##
+##           See the GNU General Public License for more details.          ##
+##    You should have received a copy of the GNU General Public License    ##
+##   along with this program. if not, see <https://www.gnu.org/licenses/>. ##
+#############################################################################
+
+"""
+Script containing validation histograms on the candidate granularity
+"""
+# pylint: disable=too-many-lines
+# pylint: disable=import-error, no-name-in-module
+from utilities_plot import buildbinning, makefill1dhist, makefill2dhist
+
+
+def fill_validation_candidates(df_reco):
+    """
+    Create histograms for the validation on the event level as a function of the multiplicity
+    """
+    _ = len(df_reco)
+
+    # Binning definition
+    binning_nsigma = buildbinning(2000, -100, 100)
+    binning_pt = buildbinning(100, 0, 100)
+
+    # Make and fill histograms
+    hlist = []
+    df_src = None
+
+    def make_and_fill(binx, namex, biny=None, namey=None, tag=""):
```

Comment:
Why is this not an argument of the function? It might also be a bit confusing to set `df_src = None` and later it's set `df_src = df_reco`. Can you, in that case, not use `df_reco` directly?

Reply:
Yes, it was a bit non intuitive, now I switched to some more integrated way in `validation.py`